File: invest-api/Contracts/BondsReportRequest.cs
namespace invest_api.Contracts
{
  /// <summary>
  /// Запрос на отчет облигаций
  /// </summary>
  public class BondsReportRequest
  {
    /// <summary>
    /// Валюта.
    /// </summary>
    public string Currency { get; set; }

    /// <summary>
    /// Дата погашения.
    /// </summary>
    public DateTime CloseDate { get; set; }

    /// <summary>
    /// Прибыль в год.
    /// </summary>
    public double Profit { get; set; }

    /// <summary>
    /// Цена дней назад.
    /// </summary>
    public double LagDay { get; set; }
  }
}

File: src/app/Models/BondsReportRequest.ts
export class BondsReportRequest {

    public Currency: string = "";
    public CloseDate:  Date | null = null;
    public Profit: number = 0;
    public LagDay: number = 0;

    constructor(currency: string, closeDate: Date, profit: number, lagDay: number ) {
        this.Currency = currency;
        this.CloseDate = closeDate;
        this.Profit = profit;
        this.LagDay = lagDay;
    }
}